### PR TITLE
rm linkback for noindex unless librarian

### PR DESCRIPTION
**Pull request by mekarpeles**
- fixes printdisabled daisy links
- adds linkback option to BookPreview (don't visibly render archive.org linkback in preview if noindex)
- shows ocaid if privileged_user (for librarians tasks)

### Screenshot

Incognito for no-index book:
testing openlibrary org_1337_books_OL24553448M_Medical_error
testing openlibrary org_1337_books_OL24553448M_Medical_error (2)

When logged in as privileged user:
testing openlibrary org_1337_books_OL24553448M_Medical_error (1)

Incognito for indexed book:
testing openlibrary org_1337_books_OL7664065M_The_Hitchhiker%27s_Guide_to_the_Galaxy

### Stakeholders

@jimchamp @seabelis

### CODE REVIEW

**Comment by jimchamp** [comment]
Is there also supposed to be a change to openlibrary/templates/type/edition/view.html in this PR?

File: openlibrary/macros/LoanStatus.html [hunk 1]
```diff
@@ -112,13 +114,13 @@
   </div>
 
 $if ocaid and secondary_action and availability.get('is_printdisabled'):
-  $:macros.BookPreview(ocaid)
+  $:macros.BookPreview(ocaid, linkback=not no_index and not is_privileged_user)
```
[the remark this answers is not in the document]
Comment (mekarpeles) [reply openlibrary/macros/LoanStatus.html:117]: I guess it's not as relevant here -- let's change it to `no no_index`. So long as the `ocaid` appears in the IDs table for librarians, I think that's sufficient

[the remark this answers is not in the document]
Comment (mekarpeles) [reply openlibrary/macros/LoanStatus.html:117]:
```suggestion
  $:macros.BookPreview(ocaid, linkback=not no_index)
```

File: openlibrary/macros/LoanStatus.html [hunk 2]
```diff
@@ -14,6 +14,8 @@
 $ availability = (doc.availability or {}) if hasattr(doc, 'availability') else {}
 $ ocaid = doc.get('ocaid') or availability.get('identifier')
 $ work_key = work_key or (doc.get('works') and doc.works[0].key)
+$ is_privileged_user = ctx.user and (ctx.user.is_admin() or ctx.user.is_librarian())
```
Comment (mekarpeles) [inline openlibrary/macros/LoanStatus.html:17]:
```suggestion
```

File: openlibrary/macros/LoanStatus.html [hunk 3]
```diff
@@ -112,13 +113,13 @@
...
 $if ocaid and secondary_action and availability.get('is_printdisabled'):
-  $:macros.BookPreview(ocaid)
+  $:macros.BookPreview(ocaid, linkback=not no_index)
   $:macros.BookSearchInside(ocaid)
 
 $:post
 
 $if ocaid and daisy:
-  $:macros.daisy('/ia/%s/daisy' % ocaid, protected=availability.get('is_printdisabled'))
+  $:macros.daisy('%s/daisy' % doc.url(), protected=availability.get('is_printdisabled'))
```
Comment (shanamatthews) [inline openlibrary/macros/LoanStatus.html:122]: @mekarpeles 

was this a typo? it looks like this is what's breaking author pages at the moment.

```suggestion
  $:macros.daisy('%s/daisy' % doc.url, protected=availability.get('is_printdisabled'))
```

**Review by shanamatthews — COMMENTED** [review]
@mekarpeles - left a comment 
just poking around here, trying to figure out what's up with author pages being broken on local builds... 😄

File: openlibrary/macros/LoanStatus.html (hunk 3 above)
Comment (cdrini) [reply openlibrary/macros/LoanStatus.html:122]: Great catch @shanamatthews ! Yep; looks like it's also breaking search pages :P

Comment (mekarpeles) [reply openlibrary/macros/LoanStatus.html:122]: We may have do dynamically compose this url in a more sophisticated way. Removing the `()` will break the contract for editions and works. Unfortunately, this "method" presents itself inconsistently across Open Library things, so the approach of using `doc.url` / `doc.url()` doesn't seem ideal for what we're trying to achieve. Presumably one of these is a property and the  other an instance method, so there also isn't a great way to bring them into alignment without more work than it's probably worth. Instead, we should create an endpoint, similar to `https://openlibrary.org/ia/borrow/{ocaid}?type=daisy` which resolves to the daisy on archive.org and sidestep the unneeded complexity.

**Comment by martiniturbide** [comment]
Hi. I got the surprise on the OpenLibrary that the option to include a IA Link on books had been removed:
NoIA_IDNumber

- @mekarpeles Remove IA link for noindex unless librarian (#5534)

If there was some legal problem or spam problem with that, maybe a different approach should be taken, like allowing the user to submit the IA Link, but have some approval process queue by a librarian, but not completely removing the possibility of the community to include a book on the openlibrary webpage. 

Or the process changed in some other way, there is some workarround, and I'm missing something ?

Regards

**Pull request merged**